src/streaming.ts

```diff
@@ -115,12 +119,20 @@ export class StreamProcessor {
       this.log.debug('SSE already connected, skip retry');
       return;
     }
+    // Cleanup the previous AbortController instance if it exists
+    if (this.abortController) {
+      this.abortController.abort();
+    }
+    this.abortController = new AbortController();
+    const { signal } = this.abortController;
 
     const isSecure = url.startsWith('https:');
     this.log.debug('SSE HTTP start request', url);
 
+    const appendedOptions = { ...options, signal };
```

Comment:
We might have an issue with this approach... officially we tell customers we support Node 12+ with this SDK, but the abort signal and `AbortController` was introduced in Node 15 https://developer.mozilla.org/en-US/docs/Web/API/AbortController#browser_compatibility. We may need to either feature detect or change our support matrix to be 15+. The reason you were getting a typescript issue was that the `@types/node` is set as `14`, so bumping that to `15` will allow `signal` in the options